src/controller/python/OpCredsBinding.cpp

```diff
@@ -319,6 +322,68 @@ void pychip_OnCommissioningStatusUpdate(chip::PeerId peerId, chip::Controller::C
     return sTestCommissioner.OnCommissioningStatusUpdate(peerId, stageCompleted, err);
 }
 
+/**
+ * By using pychip_OpCreds_AllocateControllerForCustomCommissioningFlow, we will not using the commissioning flow from the
+ * Controller class.
+ *
+ */
+PyChipError pychip_OpCreds_AllocateControllerForCustomCommissioningFlow(chip::Controller::DeviceCommissioner ** outDevCtrl,
+                                                                        uint8_t * serializedEphemeralKey,
+                                                                        uint32_t serializedEphemeralKeyLen, uint8_t * noc,
+                                                                        uint32_t nocLen, uint8_t * icac, uint32_t icacLen,
+                                                                        uint8_t * rcac, uint32_t rcacLen,
+                                                                        chip::VendorId adminVendorId, bool enableServerInteractions)
+{
+    ChipLogDetail(Controller, "Creating New Device Controller");
+
+    auto devCtrl = std::make_unique<chip::Controller::DeviceCommissioner>();
+    VerifyOrReturnError(devCtrl != nullptr, ToPyChipError(CHIP_ERROR_NO_MEMORY));
+
+    chip::Crypto::P256Keypair ephemeralKey;
+    chip::Crypto::P256SerializedKeypair keyPair;
+    memcpy(keyPair.Bytes(), serializedEphemeralKey, serializedEphemeralKeyLen);
```

Comment:
Please do not directly use serialized keypair here. The caller has to manage the P256Keypair

Reply:
already resolved.